README.md

```diff
@@ -2,48 +2,107 @@
-## Usage:
+The exporter is a stateless HTTP endpoint. On each HTTP request, it locally inspects the cluster status by parsing pre-existing distributed data, provided by the tools of the various cluster components.
 
-You can find the RPM pkgs for the exporter here: https://build.opensuse.org/package/show/server:monitoring/prometheus-ha_cluster_exporter.
+Exported data include:
+- Pacemaker cluster summary, nodes and resources stats 
+- Corosync ring errors and quorum votes
+- SBD devices health status 
+- DRBD resources and connections stats
 
-Once installed run the exporter **inside a cluster node** with: 
+A comprehensive list of all the metrics can be found in the [metrics document](doc/metric_spec.md).
 
-`systemctl start prometheus-ha_cluster_exporter`. By default it will show on `http://YOUR_HOST_IP:9002/metrics`.
+## Installation
 
-If you open a web-browser it will serve the metrics. 
-The exporter can't work outside a HA cluster node.
+The project can be installed in many ways, including but not limited to:
+
+1. [Manual clone & build](#manual-clone-&-build)
+2. [Go](#go)
+3. [RPM](#rpm)
+
+
+### Manual clone & build
+
+```
+git clone https://github.com/ClusterLabs/ha_cluster_exporter
+cd ha_cluster_exporter
+make
+make install
+```
+
+### Go
+
+```
+go get github.com/ClusterLabs/ha_cluster_exporter
+```
+
+### RPM
+You can find the repositories for RPM based distributions in [SUSE's Open Build Service](https://build.opensuse.org/repositories/server:monitoring/prometheus-ha_cluster_exporter).  
+On openSUSE or SUSE Linux Enterprise you can just use the `zypper` system package manager:
+```shell
+export DISTRO=SLE_15_SP1 # change as desired
+zypper addrepo https://download.opensuse.org/repositories/server:/monitoring/$DISTRO/server:monitoring.repo
+zypper install prometheus-ha_cluster_exporter
+```
+
+## Usage
+
+You can run the exporter in any of the cluster nodes. 
+
+```
+$ ./ha_cluster_exporter  
+INFO[0000] Serving metrics on 0.0.0.0:9002
+```
+
+Though not strictly required, it is _strongly_ advised to run it in all the nodes.
+
+It will export the metrics under the `/metrics` path, on port `9002` by default.
+
+While the exporter can run outside a HA cluster node, it won't export any metric it can't collect; e.g. it won't export DRBD metrics if it can't be locally inspected with `drbdsetup`.  
+A warning message will inform the user of such cases.
 
 **Hint:**
-For a terraform deployment you can also read: https://github.com/SUSE/ha-sap-terraform-deployments
+You can deploy a full HA Cluster via Terraform with [SUSE/ha-sap-terraform-deployments](https://github.com/SUSE/ha-sap-terraform-deployments).
+
+### Configuration
+
+All the runtime parameters can be configured either via CLI flags or via a configuration file, both or which are completely optional.
+
+For more details, refer to the help message via `ha_cluster_exporter --help`.
+
+**Note**:
+the built-in defaults are tailored for the latest version of SUSE Linux Enterprise.
```

Comment:
should we also add `openSUSE` here? :thinking: since I think should just work

Reply:
I thought about this, but we have never actually tested the project on openSUSE.

Reply:
we can safe put openSUSE here.